app/controllers/application_controller.rb

```diff
@@ -1,4 +1,35 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
+  # skip_before_action :verify_authenticity_token
+  helper_method :login!, :logged_in?, :current_user, :authorized_user?, :user_name?, :logout!, :set_user
+
+  def login!
+    session[:user_id] = @user.id
+  end
+
+  def logged_in?
+    !!session[:user_id]
```

Comment:
I had to look up the use of a double exclamation mark. You're converting the output to a guaranteed boolean. That's really clever way to get around rubys truthyness, I hadn't seen that before.

Reply:
Yeah, definitely handy